Marlin/src/module/planner.cpp

```diff
@@ -794,18 +794,12 @@ block_t* Planner::get_current_block() {
  */
 void Planner::calculate_trapezoid_for_block(block_t * const block, const_float_t entry_factor, const_float_t exit_factor) {
 
-  uint32_t initial_rate = LROUND(block->nominal_rate * entry_factor),
-           final_rate = LROUND(block->nominal_rate * exit_factor); // (steps per second)
+  const uint32_t initial_rate = _MAX(LROUND(block->nominal_rate * entry_factor), MINIMAL_STEP_RATE),
+                   final_rate = _MAX(LROUND(block->nominal_rate * exit_factor),  MINIMAL_STEP_RATE); // (steps per second)
 
-  // Limit minimal step rate (Otherwise the timer will overflow.)
-  NOLESS(initial_rate, uint32_t(MINIMAL_STEP_RATE));  // Enforce the minimum speed
-  NOLESS(final_rate, uint32_t(MINIMAL_STEP_RATE));
+  // Now ensure the nominal rate is above minimum
   NOLESS(block->nominal_rate, MINIMAL_STEP_RATE);
 
-  // Assume these are handled elsewhere. Retained for debugging.
-  //NOMORE(initial_rate, block->nominal_rate);        // NOTE: The nominal rate may be less than MINIMAL_STEP_RATE!
-  //NOMORE(final_rate, block->nominal_rate);
-
```

Comment:
Following up on this PR: I think this version of the code is the most readable and easiest for the follow up PR #27035.

```C++
  uint32_t initial_rate = LROUND(block->nominal_rate * entry_factor),
           final_rate = LROUND(block->nominal_rate * exit_factor); // (steps per second)

  // Limit minimal step rate (Otherwise the timer will overflow.)
  NOLESS(initial_rate, MINIMAL_STEP_RATE);
  NOLESS(final_rate, MINIMAL_STEP_RATE);
  NOLESS(block->nominal_rate, MINIMAL_STEP_RATE);
```

Reply:
I like keeping the limiting code as a separate block for readability as well.

Reply:
I'm reluctant to change it. Maybe that's a character flaw.

Stylistically, I tend to make values like `initial_rate` and `final_rate` that are calculated but then never altered `const` so that the compiler can catch attempts to modify or reuse them, either intentionally or due to a typo. And I find the use of `_MAX` in the assignment clearer because it's right in the assignment. It's clear from the single assignment line that the value has been limited and that it's not going to be changing later.

Questions of style are always a bit tricky. Our basic guideline is to be clear but not pedantic. My main aim here was to get that `const` specifier. There may be some small sacrifice in alignment to the limiting of `block->nominal_rate`, but that line should actually stand out because it's the first step in modifying the block.

I've been pondering this question for over 15 minutes, so let that be a character note! : If you really want to derail my day just initiate a debate about coding style.

Reply:
> I've been pondering this question for over 15 minutes, so let that be a character note!

Agonising over style is a good character trait IMHO. And agonising for 15 minutes is probably about the right amount of time if you ever  actually want to get things done.